src/clarity-angular/data/datagrid/render/dom-adapter.ts

```diff
@@ -10,13 +10,19 @@
  * to native DOM elements' methods and properties in the Datagrid happens here.
  */
 
-import {Injectable} from "@angular/core";
+import {isPlatformBrowser} from "@angular/common";
+import {Inject, Injectable, PLATFORM_ID} from "@angular/core";
 
 @Injectable()
 export class DomAdapter {
+    constructor(@Inject(PLATFORM_ID) private platformId: Object) {}
+
     userDefinedWidth(element: any): number {
         element.classList.add("datagrid-cell-width-zero");
-        const userDefinedWidth = parseInt(getComputedStyle(element).getPropertyValue("width"), 10);
+        let userDefinedWidth = 0;
+        if (isPlatformBrowser(this.platformId)) {
```

Comment:
Guarding every single method of the class is a bit painful. I think a cleaner solution would be to have a separate `NoopDomAdapter` or other dummy name, that pretty much returns 0 for everything, and then you switch the provider declaration from `providers: [DomAdapter]` to:
```typescript
providers: [{
    provide: DomAdapter,
    useFactory: function(platformId: Object) {
        if (isPlatformBrowser(platformId)) {
            return new DomAdapter();
        } else {
            return new NoopDomAdapter();
        }
    },
    deps: [PLATFORM_ID]
}]
```

I think it's both more maintainable and flexible at the same time. If we ever were to have an actual implementation on the server-side, we would already have a clean separate class for it.
That was actually the point of isolating all DOM access for the Datagrid to a single class.

Reply:
Sure, will do.